share/shwap/sample.go

```diff
@@ -0,0 +1,136 @@
+package shwap
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/nmt"
+	nmt_pb "github.com/celestiaorg/nmt/pb"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	pb "github.com/celestiaorg/celestia-node/share/shwap/proto"
+)
+
+// Sample represents a data share along with its Merkle proof, used to validate the share's
+// inclusion in a data square.
+type Sample struct {
+	share.Share             // Embeds the share which includes the data with namespace.
+	Proof       *nmt.Proof  // Proof is the Merkle Proof validating the share's inclusion.
+	ProofType   rsmt2d.Axis // ProofType indicates whether the proof is against a row or a column.
```

Comment:
Let's call it Axis. IMO, ProofType refers to the different type of proof as a data structure.